hmrc-mongo-play-27/src/main/scala/uk/gov/hmrc/mongo/play/json/CollectionFactory.scala

```diff
@@ -29,11 +32,16 @@ trait CollectionFactory {
     collectionName: String,
     domainFormat: Format[A]
   ): MongoCollection[A] =
-   db.getCollection[A](collectionName)
+    db.getCollection[A](collectionName)
       .withCodecRegistry(
         CodecRegistries.fromRegistries(
-          CodecRegistries.fromCodecs(Codecs.playFormatCodec(domainFormat)),
-          DEFAULT_CODEC_REGISTRY))
+          CodecRegistries.fromCodecs(
+            Codecs.playFormatCodec(domainFormat),
+            new UuidCodec(UuidRepresentation.STANDARD)
```

Comment:
I'm a bit concerned about registering a uuid codec where we hadn't before, and setting the type to Standard. I appreciate that without it, we never would have supported Uuids directly in queries/updates so it's not breaking anything per se, but wonder if we should be providing some flexibility, e.g. someone has migrated a repo where they previously stored the data in Legacy format. (It would break using UUID in queries/filters if stored in Legacy format)

Should we add a `extraCodecs` param (and similarly to `PlayMongoRepository`) so it can be provided by the client? We've seen many cases where this could be useful in general.

Reply:
Yeah I have definitely considered opening a PR to add `extraCodecs: Codec[_]*` to the CollectionFactory before, in fact I started to do it on this PR but didn't want to put two unrelated changes in 😆

Reply:
We initially had this in `CollectionFactory` and `PlayMongoRepository`, but removed it since it was more cumbersome than explicitly using `Codec.toBson/fromBson` but it seems that it is useful.
Here it could solve us having to pick the format to support.
I have no idea what would happen if we registered both legacy and standard, could it help read both formats - how would it know which to write? I expect it would fail. In which case, how would we migrate data from old to new format?

Reply:
It's very awkward to do the migration since you can only have one codec registered for encoding UUIDs in Filters and Updates, and also the `UuidHelper` refuses to read `JAVA_LEGACY` data when configured for `STANDARD` mode

Reply:
I have added this `extraCodecs` now